pghoard/pghoard.py

```diff
@@ -351,8 +373,19 @@ def handle_site(self, site, site_config):
 
         chosen_backup_node = random.choice(site_config["nodes"])
 
-        if site not in self.receivexlogs and site_config["active_backup_mode"] == "pg_receivexlog":
-            self.receivexlog_listener(site, chosen_backup_node, xlog_path + "_incoming")
+        if site not in self.receivexlogs and site not in self.walreceivers:
+            if site_config["active_backup_mode"] == "pg_receivexlog":
+                self.receivexlog_listener(site, chosen_backup_node, xlog_path + "_incoming")
+            elif site_config["active_backup_mode"] == "walreceiver":
+                state_file_path = self.config["json_state_file_path"]
+                walreceiver_state = {}
+                if os.path.exists(state_file_path):
+                    old_state_file = config.read_json_config_file(state_file_path)
```

Comment:
I don't think `read_json_config_file()` is appropriate for reading the state file.